### PR TITLE
added cli options to output ACLs

### LINKED ISSUE #492: Print out loaded ACLs
Print out loaded ACLs. This would be awesome to make sure all ACLs are loaded.

### PR DESCRIPTION
**Pull request by samson4649**
- [x] read the [CONTRIBUTING guidelines](README.md#user-content-contributing)
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [ ] updated documentation if needed
- [x] updated CHANGELOG.md

Drafted to Fix #492

### CODE REVIEW

**Comment by samson4649** [comment]
Requires integration testing with separate instances of headscale per test or by dynamically loading ACLs into a single running instance (issue #582 )

File: cmd/headscale/cli/acls.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		if err != nil {
+			ErrorOutput(
+				err,
+				fmt.Sprintf("Error getting headscale app: %s", err),
+				output,
+			)
+
+			return
+		}
+		policy := h.GetACLPolicy()
```
Comment (restanrm) [inline cmd/headscale/cli/acls.go:40]: This won't do, you need to go through the protobuf API for this. Add message type and response in proto folder, generate code via `make generate` then edit `grpcv1.go` to add server-side part

**Review by restanrm — CHANGES_REQUESTED** [review]

**Comment by samson4649** [comment]
@restanrm - as discussed, changes have been made to implement this over the grpc server.

**Comment by restanrm** [comment]
It behaves as expected what do you think of this @kradalby ?

**Comment by samson4649** [comment]
This still needs some things fixed up - still working on it

**Comment by samson4649** [comment]
g2g now - ready for review

**Comment by kradalby** [comment]
Hi, I am happy to give this a go, apologise for the delay, can you do a pass through the linters (proto, golangci and prettier) and then we can merge?

And update the hash in `flake.nix` (see the fail build)

**Comment by juanfont** [comment]
Hey @samson4649, apologies for the delay. Can you check the conflicts?

**Comment by juanfont** [comment]
@samson4649 can you check the (minor) conflicts? I think we are pretty ready to merge.

**Comment by juanfont** [comment]
@samson4649 we have the linters working again, and found a couple of minor things here.

There is also an issue with a dependency. Can you have a look? Thanks!!

File: go.mod [hunk 2]
```diff
@@ -30,15 +30,15 @@ require (
 	golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa
 	golang.org/x/oauth2 v0.0.0-20220808172628-8227340efae7
 	golang.org/x/sync v0.0.0-20220722155255-886fb9371eb4
-	google.golang.org/genproto v0.0.0-20220808204814-fd01256a5276
```
Comment (juanfont) [inline go.mod:33]: Do not downgrade, please

File: go.mod [hunk 3]
```diff
@@ -30,15 +30,15 @@ require (
...
+	google.golang.org/genproto v0.0.0-20220810155839-1856144b1d9c
 	google.golang.org/grpc v1.48.0
 	google.golang.org/protobuf v1.28.1
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
 	gorm.io/driver/postgres v1.3.8
 	gorm.io/gorm v1.23.8
-	inet.af/netaddr v0.0.0-20220617031823-097006376321
```
Comment (juanfont) [inline go.mod:41]: Downgrade :(

File: go.mod [hunk 4]
```diff
@@ -70,7 +70,7 @@ require (
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510 // indirect
 	github.com/google/uuid v1.3.0 // indirect
 	github.com/gookit/color v1.5.0 // indirect
-	github.com/hashicorp/go-version v1.4.0 // indirect
```
Comment (juanfont) [inline go.mod:73]: Same

File: acls.go [hunk 5]
```diff
@@ -123,6 +124,144 @@ func (h *Headscale) UpdateACLRules() error {
 	return nil
 }
 
+func (h *Headscale) ListACLPolicy() (*ACLPolicy, error) {
+	return h.aclPolicy, nil
+}
+
+func ACLProtoToStruct(v *v1.ACLPolicy) (*ACLPolicy, error) {
```
Comment (juanfont) [inline acls.go:131]: Here the linter is complaining

> parameter name 'v' is too short for the scope of its usage (varnamelen)


Replace `v`for `policy` or something like this, for instance.

File: acls.go [hunk 6]
```diff
@@ -123,6 +124,144 @@ func (h *Headscale) UpdateACLRules() error {
...
+		protoTagOwners[k] = protoTagOwner
+	}
+	return protoTagOwners
+}
+
+func (h *Hosts) toProto() map[string]string {
+	protoHosts := make(map[string]string, len(*h))
+	for k, v := range *h {
+		protoHosts[k] = v.String()
+	}
```
Comment (juanfont) [inline acls.go:261]:
```suggestion
	}
	
```

File: acls.go [hunk 7]
```diff
@@ -123,6 +124,144 @@ func (h *Headscale) UpdateACLRules() error {
...
+	return protoHosts
+}
```
Comment (juanfont) [inline acls.go:263]:
```suggestion
}

```

File: cmd/headscale/cli/acls.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+			return
+		}
+
+		SuccessOutput(
+			policy,
+			``,
+			output,
+		)
+
+		return
```
Comment (juanfont) [inline cmd/headscale/cli/acls.go:76]:
```suggestion
```

Comment (juanfont) [reply cmd/headscale/cli/acls.go:76]: not required

**Review by juanfont — COMMENTED** [review]
In addition to those comments, if you make changes to go.mod and go.sum, you need to update its checksum in `flake.nix`.

**Pull request closed without merging**